packages/ida-auth-sdk/src/mosip-authenticator.ts

```diff
@@ -0,0 +1,129 @@
+import {
+  encryptAuthData,
+  extractKeysFromPkcs12,
+  getPemCertificateThumbprint,
+  signAuthRequestData,
+  urlSafeCertificateThumbprint,
+} from "@opencrvs/mosip-crypto";
+import fs from "node:fs";
+import crypto from "node:crypto";
+
+interface MOSIPAuthenticatorConfig {
+  partnerApiKey: string;
+  partnerMispLk: string;
+  partnerId: string;
+  idaAuthDomainUri: string;
+  idaAuthUrl: string;
+  encryptCertPath: string;
+  decryptP12FilePath: string;
+  decryptP12FilePassword: string;
+  signP12FilePath: string;
+  signP12FilePassword: string;
+}
+
+interface AuthParams {
+  individualId: string;
+  individualIdType: string;
+  demographicData: { dob: string };
+  consent: boolean;
+}
+
+export default class MOSIPAuthenticator {
```

Comment:
Yes, it's a `class`. Wanted to keep the implementation similar to `ida-auth-sdk` from mosip.